part-4/modularization-concept.md

```diff
@@ -0,0 +1,248 @@
+
+    When creating a simulation, these events are created separately based on their container criteria. Therefore, the simulation will contain two events called `Event 1` - one event in the node `Events` with the parameter `Param 1`, and one event in the node `Organism` with the parameter `Param 2`, disregarding the merge behavior.
+
+2. If the event from module `B` has container criteria `Events` OR `Organism`
+    - Merge behavior **Overwrite**: the final model will contain the `Event 1` in both containers `Events` and `Organism` with `Param 2` only (`Organism|Event 1` is defined in module `B` only, and `Events|Event 1` is defined in both modules and will be overwritten by the event defined in module `B`).
+    - Merge behavior **Extend**: the final model will contain the `Event 1` in container `Organism` with the parameter `Param 1`, and the event `Event 1` in container `Events` with parameters `Param 1` and `Param 2` (`Events|Event 1` is defined in both modules and will be extended).
+
+{% hint style="warning" %}
+The container criteria of events are not combined in any way. The events across different modules are generated separately based on their container criteria and combined (merge or extend) only if they are created in the same container.
+{% endhint %}
+
+{% hint style="note" %}
+This behavior will be changed in a future release, allowing more straightforward combination of events across modules.
+{% endhint %}
+
+For the events with identical final absolute paths (generated by the container criteria), the following rules apply:
+
+#### Merge behavior "Overwrite"
+
+The complete tree structure of the event is overwritten. This means:
+
+- **Parameters** list is overwritten. Only parameters defined in the module that is lower in the hierarchy are used.
+
+- **Administered molecule** is overwritten.
+
+- All subnodes of the event (e.g., **ProtocolSchemaItem**, **Application_StartEvent**, etc.) are overwritten.
+
+For each event:
+
+- **Events start condition** equation is overwritten.
+- **Events start condition - "One Time" checkbox** is overwritten.
+- The list of **Assignments** is overwritten.
+
+#### Merge behavior "Extend"
+
+The tree structure of the event is extended. This means:
+
+ - **Parameters** list is extended. If the same parameter is defined in multiple modules, the parameter from the module that is lower in the hierarchy is used.
+
+- **Administered molecule** is extended.
+
+{% hint style="warning" %}
+ This results in a malformed event if different molecules are defined in different modules!
+{% endhint %}
+
+For each event:
+
+- **Events start condition** equation: Changes are not applied!
+- **Events start condition - "One Time" checkbox**: Changes are not applied!
```

Comment:
Really? IF yes, I think we should change it (after 12.1). Otherwise, the logic is so hard to understand and to memorize...

Reply:
Yes of course, the whole logic should be re-evaluated.